crates/ruff_linter/src/rules/pycodestyle/rules/invalid_escape_sequence.rs

```diff
@@ -74,7 +74,7 @@ pub(crate) fn invalid_escape_sequence(
             let Some(range) = indexer.fstring_ranges().innermost(token_range.start()) else {
                 return;
             };
-            (value.as_str(), range.start())
+            (value.as_ref(), range.start())
```

Comment:
I don't think you want `.as_ref()` here. Generally one should only use `as_ref()` when a `AsRef` bound is in scope, otherwise you're susceptible to inference failures in future Rust releases.

If `value` is a `Box<str>`, then you can get a `&str` with `&*value`. If `value` is a `&Box<str>` (which is I think the case here), then you can get a `&str` with `&**value`.

Reply:
And I would generally apply this same comment to I believe all other uses of `as_ref()` in this PR.

Reply:
Some references:

* https://github.com/rust-lang/rust/issues/60958
* https://github.com/rust-lang/rust/issues/62586

Reply:
Makes sense. I like `as_ref` because I don't need to guess the number of `*` that I have to use 😆 but I can change it. 

I'm somewhat hesitant of merging the PR because I don't see an improvement when running the lexer benchmarks locally :(

Reply:
I do see a very small improvement on my machine:

```
$ critcmp base test
group                        base                                   test
-----                        ----                                   ----
lexer/large/dataset.py       1.01    202.0±0.66µs   201.4 MB/sec    1.00    200.9±0.51µs   202.5 MB/sec
lexer/numpy/ctypeslib.py     1.03     48.9±0.15µs   340.5 MB/sec    1.00     47.7±0.17µs   349.2 MB/sec
lexer/numpy/globals.py       1.04      5.0±0.02µs   592.9 MB/sec    1.00      4.8±0.01µs   616.2 MB/sec
lexer/pydantic/types.py      1.02     93.8±0.25µs   271.9 MB/sec    1.00     92.2±0.37µs   276.6 MB/sec
lexer/unicode/pypinyin.py    1.01     12.5±0.03µs   335.4 MB/sec    1.00     12.4±0.05µs   337.6 MB/sec
```

I'd say ship it!